src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -233,12 +233,10 @@ public ExamResource(UserRepository userRepository, CourseRepository courseReposi
             else {
                 exam = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(examId);
             }
-            examService.setExamExerciseProperties(exam);
+            examService.setExamProperties(exam);
             return ResponseEntity.ok(exam);
         }
         Exam exam = examRepository.findByIdWithRegisteredUsersElseThrow(examId);
-        examRepository.setNumberOfRegisteredUsersForExams(Collections.singletonList(exam));
```

Comment:
Why is this now only done if `withExerciseGroups` is true (was not done for `!withStudents && !withExerciseGroups` as well, so might not be a problem)?

Reply:
yeah we currently only use in for these. We removed the withStudents option in this case from the call as its not needed in this case.